src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
yeah I was pondering this earlier. I think it might be a decent idea to hook up that [signal](http://doc.qt.io/qt-5/qobject.html#destroyed) to go through and clean the list. This is more of a consideration for longer-running servers, and I think perhaps maybe one worth pursuing. Thoughts?